app/src/main/java/com/example/dine_and_donate/Activities/HomeActivity.java

```diff
@@ -35,21 +50,33 @@
     private Fragment mProfileFragment = new ProfileFragment();
     public User mCurrentUser;
 
+    private Fragment mDefaultFragment;
+    public LatLng markerLatLng;
+
+    private PendingIntent pendingIntentam;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_home);
 
         mDrawerNav = findViewById(R.id.drawerNav);
         mCurrentUser = Parcels.unwrap(getIntent().getParcelableExtra(User.class.getSimpleName()));
+        mDefaultFragment = (getIntent().getStringExtra("defaultFragment") != null) ? mMapFragment : mProfileFragment;
+        String latitude = getIntent().getStringExtra("latitude");
+        String longitude = getIntent().getStringExtra("longitude");
+        markerLatLng = (latitude != null || longitude != null) ? new LatLng(Double.parseDouble(latitude), Double.parseDouble(longitude)) : markerLatLng;
```

Comment:
this should probably be && not || right? both of them need to be nonnull? also, did you mean to set markerLatLng to itself?